src/app.js

```diff
@@ -26,52 +26,55 @@ function main(sources) {
     time: new Date()
   }]);
 
+  const intent = DOM => {
+    const changeMessage$ = sources.DOM
+      .select('.input-message')
+      .events('input')
+      .map(e => e.target.value)
+      .share();
```

Comment:
So we consume `changeMessage$` in multiple places, but we only ever want to consume its latest value once, so therefore we need the share, else both the consumers of `changeMessage$` would run every time there's a new value, and we don't want that?